src/main/java/cholog/wiseshop/api/member/service/MemberService.java

```diff
@@ -0,0 +1,43 @@
+package cholog.wiseshop.api.member.service;
+
+import cholog.wiseshop.api.member.dto.request.SignInRequest;
+import cholog.wiseshop.api.member.dto.request.SignUpRequest;
+import cholog.wiseshop.db.member.Member;
+import cholog.wiseshop.db.member.MemberRepository;
+import jakarta.servlet.http.HttpSession;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional
+public class MemberService {
+
+    private static final String SESSION_KEY = "member";
+    private final MemberRepository memberRepository;
+    private final PasswordEncoder passwordEncoder;
+
+    public MemberService(MemberRepository memberRepository, PasswordEncoder passwordEncoder) {
+        this.memberRepository = memberRepository;
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    public void signUpMember(SignUpRequest signUpRequest) {
+        String encodePassword = passwordEncoder.encode(signUpRequest.password());
+        Member member = new Member(signUpRequest.email(), signUpRequest.name(), encodePassword);
+        if (memberRepository.findByEmail(signUpRequest.email()).isPresent()) {
+            throw new IllegalArgumentException("이미 존재하는 회원입니다.");
+        }
+        memberRepository.save(member);
+    }
```

Comment:
```suggestion
    public void signUpMember(SignUpRequest signUpRequest) {
      if (memberRepository.findByEmail(signUpRequest.email()).isPresent()) {
              throw new IllegalArgumentException("이미 존재하는 회원입니다.");
          }
        String encodePassword = passwordEncoder.encode(signUpRequest.password());
        Member member = new Member(signUpRequest.email(), signUpRequest.name(), encodePassword);
        memberRepository.save(member);
    }
```
이미 존재하는 회원인지 검증하는 코드를 제일 상단에 두면, 예외가 발생했을 때, `Member` 객체를 생성하지 않아도 될 것 같습니다.

Reply:
오~! 그러네요!!
불필요한 객체를 생성하지 않아도 되겠어요
반영했습니다 감사합니당🐥